doc/communicationDiagram.md

```diff
@@ -2,76 +2,98 @@
 
 ## Overview
 
-The Communication Diagram shows how objects interact in TcpServer, along with messages that travel from one to another.
+The Communication Diagram shows how internal objects interact in **TcpServer** top-level class, along with messages that travel from one to another.
 
 ## Diagram Conventions
 
-We will use PlantUML to draw the Communication Diagram, but it does not support this type of diagram, so we will adapt it to get what we need.
-We will consider objects as the VIs in yellow boxes, with a label `[object:class]`; the messages as user events with red arrows, labeled 1:message1, 2:message2, 3:message3, etc., where the numerical prefix to the message name indicates its user event reference id in the sequence; and a yellow label at the left of the boxes with the id and name of user events.
+The [PlantUML](https://plantuml.com) is used to draw the Communication Diagram.
+Although it does not support this diagram natively, it is still possible to emulate the expected behavior.
+The objects with functions related to communication are in rectangle boxes.
+The arrow between objects gives the direction of communication and the related LabVIEW user event name.
 
-## User Events
+## List of User Events
 
-This is a list of user events in TcpServer, with a numeric id to indetify them in the Communication Diagram.
+The following is the list of user events in **TcpServer**:
 
 | id | User Event |
 |:--:|:----------:|
-| 1 | CmdWrite |
-| 2 | CmdRead  |
-| 3 | EvtWrite |
-| 4 | EvtRead  |
-| 5 | GenWrite |
-| 6 | GenRead  |
-
-The following table shows the user events with their publisher and subscriber.
-
-| User Event | id | Publisher | Subscriber |
-|:--------:|:----:|----------:|-----------:|
-| CmdRead  | 2 | TcpServerCmd:processCmd.vi  | ComponentSimulator.vi     |
-| CmdWrite | 1 | TcpServer:sendCmdStatus.vi  | TcpServerCmd.runServer.vi |
-| EvtRead  | 4 | TcpServerCmd:processEvt.vi  | ComponentSimulator.vi     |
-| EvtWrite | 3 | TcpServer.sendEvt.vi         | TcpServerCmd.runServer.vi |
-| GenRead  | 6 | TcpServer:getClientStatus.vi | ComponentSimulator.vi     |
-| GenWrite | 5 |                              | TcpServerCmd:runServer.vi |
-|          |   | TcpServerCmd.runServer.vi    |                           |
-
-## Top Level User Events
-
-We will define a group of user events closest to the User and we will name them `Top Level User Events`.
-These user events are what the User needs to know and use.
-They are defined in the following table.
+| 1  | CmdRead    |
+| 2  | EvtRead    |
+| 3  | GenRead    |
+| 4  | CmdWrite   |
+| 5  | EvtWrite   |
+| 6  | GenWrite   |
+
+- `CmdRead` is used to send the `tcpPacket` to the Component if the incoming command is legal.
```

Comment:
Maybe you need to explain what is the "Component" here.

Reply:
Done.